packages/origin-ui-core/src/components/devices/ProducingDeviceTable.tsx

```diff
@@ -57,20 +57,20 @@ export function ProducingDeviceTable(props: IOwnProps) {
     const user = useSelector(getUserOffchain);
     const producingDevices = useSelector(getProducingDevices);
     const baseURL = useSelector(getBaseURL);
-    const organizations = useSelector(getOrganizations);
     const environment = useSelector(getEnvironment);
+    const offChainDataSource = useSelector(getOffChainDataSource);
 
     const dispatch = useDispatch();
 
-    function enrichProducingDeviceData(): IEnrichedProducingDeviceData[] {
+    async function enrichProducingDeviceData(): Promise<IEnrichedProducingDeviceData[]> {
+        const deviceClient = offChainDataSource.deviceClient;
+        const devicesWithRelations = await deviceClient.getAll(false, false);
```

Comment:
This is redundant. You are re-fetching all the devices. Devices are already fetched and stored in the `producingDevices` variable.